### PR TITLE
build: Add cross building option

### PR DESCRIPTION
**Pull request by toshywoshy**
Added a cross build option using a buildrelease function

Signed-off-by: Toshaan Bharvani <toshaan@vantosh.com>

ping @purpleidea

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -114,6 +115,14 @@ $(PROGRAM): $(GO_FILES)
 	@echo "Building: $(PROGRAM), version: $(SVERSION)..."
 	time go build -i -ldflags "-X main.program=$(PROGRAM) -X main.version=$(SVERSION)" -o $(PROGRAM) $(BUILD_FLAGS);
 
+define buildrelease
```
Comment (purpleidea) [inline Makefile]: wow. I am always impressed when someone uses new `make` magic that I don't yet know!

Thank you wizard!

File: Makefile [hunk 2]
```diff
@@ -114,6 +115,14 @@ $(PROGRAM): $(GO_FILES)
...
+GOOS=$(1) GOARCH=$(2) go build -ldflags "-X main.program=$(PROGRAM) -X main.version=$(SVERSION) -s -w" -o $(PROGRAM)-$(1)-$(2) $(BUILD_FLAGS);
+endef
+
+cross:
```
Comment (purpleidea) [inline Makefile]: does this need an entry in `.PHONY` above too?

If you can rename this to `crossbuild` or something slightly more obvious that would be awesome as well!

File: Makefile (hunk 1 above)
Comment (toshywoshy) [reply Makefile]: Well, I cannot take full credit, it is based upon some other cross build functions I mixed up together.

File: Makefile (hunk 2 above)
Comment (toshywoshy) [reply Makefile]: Well, yes, renaming to `crossbuild` is more logical

File: Makefile [hunk 3]
```diff
@@ -16,7 +16,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 SHELL = /usr/bin/env bash
-.PHONY: all art cleanart version program path deps run race bindata generate build clean test gofmt yamlfmt format docs rpmbuild mkdirs rpm srpm spec tar upload upload-sources upload-srpms upload-rpms copr
+.PHONY: all art cleanart version program path deps run race bindata generate build clean test gofmt yamlfmt format docs rpmbuild mkdirs rpm srpm spec tar upload upload-sources upload-srpms upload-rpms copr crossbuild
```
Comment (purpleidea) [inline Makefile]: mind adding this addition in order with where crossbuild is in the file please?

**Comment by toshywoshy** [comment]
@purpleidea this should work and be good to go

**Comment by purpleidea** [comment]
@toshywoshy Can you rebase all the commits into a single patch please? Also see the above comment in https://github.com/purpleidea/mgmt/pull/295#discussion_r164755013 please.

**Comment by purpleidea** [comment]
@toshywoshy LGTM, thanks, MERGED :)

**Pull request merged**